src/sentry/static/sentry/app/components/group/pluginActions.jsx

```diff
@@ -0,0 +1,232 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import createReactClass from 'create-react-class';
+import Modal from 'react-bootstrap/lib/Modal';
+import ApiMixin from 'app/mixins/apiMixin';
+import DropdownLink from 'app/components/dropdownLink';
+import GroupState from 'app/mixins/groupState';
+import MenuItem from 'app/components/menuItem';
+import plugins from 'app/plugins';
+import {t} from 'app/locale';
+import {toTitleCase} from 'app/utils';
+import IssueSyncListElement from 'app/components/issueSyncListElement';
+
+
+const PluginActions = createReactClass({
+  displayName: 'PluginActions',
+
+  propTypes: {
+    group: PropTypes.object.isRequired,
+    plugin: PropTypes.object.isRequired,
+  },
+
+  mixins: [ApiMixin, GroupState],
+
+  getInitialState() {
+    return {
+      showModal: false,
+      actionType: null,
+      issue: null,
+      pluginLoading: false,
+    };
+  },
+
+  componentWillMount() {
+    this.loadPlugin(this.props.plugin);
+  },
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.plugin.id !== nextProps.plugin.id) {
+      this.loadPlugin(nextProps.plugin);
+    }
+  },
+
+  ACTION_LABELS: {
+    create: t('Create New Issue'),
+    link: t('Link with Existing Issue'),
+    unlink: t('Unlink Issue'),
+  },
+
+  deleteIssue() {
+    const plugin = this.props.plugin;
+    // override plugin.issue so that 'create/link' Modal
+    // doesn't think the plugin still has an issue linked
+    Object.assign(plugin, {
```

Comment:
nit: I would use object spread to combine this with the above assignment, e.g.:

```javascript
const plugin = {
  ...this.props.plugin,
  issue: null,
};
```

Reply:
so I actually want to mutate `this.props.plugin` here, I don't think the spread operator does that right?

Reply:
That's a red flag if we are relying on `this.props` being mutated, why do we need it mutated?

Reply:
to avoid this https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/plugins/components/issueActions.jsx#L48 from not calling `fetchData()`, there could be a better way to approach this, but I couldn't seem to get anything else working